sqlair.go

```diff
@@ -246,3 +246,44 @@ func (q *Query) All(sliceArgs ...any) (err error) {
 
 	return nil
 }
+
+type TX struct {
+	tx *sql.Tx
+}
+
+func (db *DB) NewTX(tx *sql.Tx) *TX {
+	return &TX{tx: tx}
+}
+
+func (db *DB) Begin(ctx context.Context, opts *sql.TxOptions) (*TX, error) {
```

Comment:
This should be our own TXOptions type, not the underlying ones. Otherwise we're stuck with that forever.

Reply:
Changed.